src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const dateParts = date.split(`${oldSeparator}`);
+  const dateObject = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObject[fromFormat[i]] = dateParts[i];
+  }
+
+  if (dateObject.YY) {
+    if (dateObject.YY < '30') {
+      dateObject.YYYY = '20' + dateObject.YY;
+    } else {
+      dateObject.YYYY = '19' + dateObject.YY;
+    }
```

Comment:
The logic for converting YY to YYYY seems to be incorrect. The condition should be based on the numeric value, not the string comparison. You should parse `dateObject.YY` as a number before comparing it to 30.